### PR TITLE
feat: Use `restorecon -T 0` on Fedora and RHEL > 8

### PR DESCRIPTION
**Pull request by bachradsusi**
From restorecon(8):

       -T nthreads
              use up to nthreads threads.  Specify 0 to create as many
              threads as there are available CPU cores; 1 to use only a
              single thread  (default); or any positive number to use
              the given number of threads (if possible).

### CODE REVIEW

**Comment by bachradsusi** [comment]
I'm not sure what's wrong with the commit message so that commitlint failed.

**Comment by bachradsusi** [comment]
[citest]

File: tasks/main.yml [hunk 1]
```diff
@@ -82,13 +82,13 @@
   with_items: "{{ selinux_fcontexts }}"
 
 - name: Restore SELinux labels on filesystem tree
-  command: /sbin/restorecon -R -F -v {{ item }}
+  command: /sbin/restorecon -R -F -v {{ restorecon_T }} {{ item }}
```
Comment (bachradsusi) [inline tasks/main.yml:85]: This should be restorecon_threads

**Comment by bachradsusi** [comment]
[citest]

**Comment by bachradsusi** [comment]
[citest]

**Comment by bachradsusi** [comment]
[citest]

File: tests/tests_restore_dirs.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        - name: Create temporary file
+          ansible.builtin.tempfile:
+            path: "{{ tempdir.path }}"
+            state: file
+            suffix: temp
+          register: tempfile
+
+        - name: Set default_t context to {{ tempdir.path }}
+          ansible.builtin.command: chcon -R -t default_t "{{ tempdir.path }}"
```
Comment (richm) [inline tests/tests_restore_dirs.yml:21]:
```suggestion
          command: chcon -R -t default_t {{ tempdir.path | quote }}
```
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_filters.html#manipulating-strings

perhaps there are other cases in the selinux role which use `command: ... "{{ some_variable }}"` but for new code it is preferred to use `{{ some_variable | quote }}` instead.

File: tests/tests_restore_dirs.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          changed_when: true
+
+        - name: Run the role
+          include_role:
+            name: linux-system-roles.selinux
+          vars:
+            selinux_fcontexts:
+              - {target: "{{ tempdir.path }}(/.*)?", setype: 'user_tmp_t'}
```
Comment (richm) [inline tests/tests_restore_dirs.yml:29]:
```suggestion
              - {target: "{{ tempdir.path }}(/.*)?", setype: "user_tmp_t"}
```
double quotes are preferred except within a `"{{ ... 'quoted string' ... }}"` Jinja expression

File: tests/tests_restore_dirs.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            selinux_restore_dirs:
+              - "{{ tempdir.path }}"
+
+        - name: Get SELinux context of {{ tempfile.path }}
+          ansible.builtin.command: ls -Z {{ tempfile.path }}
```
Comment (richm) [inline tests/tests_restore_dirs.yml:34]:
```suggestion
          command: ls -Z {{ tempfile.path | quote }}
```
use `quote` here - or not - not sure if you need to quote the use of `tempfile.path` - but if you do, please be consistent

File: tests/tests_restore_dirs.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          file:
+            path: "{{ tempdir.path }}"
+            state: absent
+        - name: Clean temporary fcontext
+          include_role:
+            name: linux-system-roles.selinux
+          vars:
+            selinux_fcontexts:
+              - {target: "{{ tempdir.path }}(/.*)?", setype: 'user_tmp_t',
+                 state: 'absent'}
```
Comment (richm) [inline tests/tests_restore_dirs.yml:54]: double quotes - see above

File: tests/tests_restore_dirs.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+- name: Check if selinux role restores directory SELinux context
+  hosts: all
+  tasks:
+    - name: Create tempdir, set fcontext and relabel
+      block:
+        - name: Create temporary directory
+          ansible.builtin.tempfile:
```
[the remark this answers is not in the document]
Comment (bachradsusi) [reply tests/tests_restore_dirs.yml:8]: I followed examples in `ansible-doc tempfile`

File: tests/tests_restore_dirs.yml (hunk 2 above)
Comment (bachradsusi) [reply tests/tests_restore_dirs.yml:21]: I guess the quotes are not necessary at all given that it's directory generated by `mktemp`

Comment (bachradsusi) [reply tests/tests_restore_dirs.yml]: fixed

File: tests/tests_restore_dirs.yml (hunk 3 above)
Comment (bachradsusi) [reply tests/tests_restore_dirs.yml]: fixed

File: tests/tests_restore_dirs.yml (hunk 4 above)
Comment (bachradsusi) [reply tests/tests_restore_dirs.yml]: fixed

**Comment by bachradsusi** [comment]
Thanks for the review. I believe I've addressed all your comments.

**Comment by bachradsusi** [comment]
[citest]

**Comment by bachradsusi** [comment]
Failing tests are probably not related to this PR.

**Pull request merged**